Lib/functools.py

```diff
@@ -994,19 +993,15 @@ def __get__(self, instance, owner=None):
             raise TypeError(msg) from None
         val = cache.get(self.attrname, _NOT_FOUND)
         if val is _NOT_FOUND:
```

Comment:
It seems like doing a check for the value is nearly useless.  The value wasn't there when `__get__` was called, and would only be there now if another thread finished before this point in the code.